### DISCUSSION TITLE
vcpkg process too slow

### DISCUSSION THREAD

**Discussion by xbotuk** [microsoft/vcpkg#24908] [Ideas]
**Describe the bug**
vcpkg process too slow takes forever since the packages are not prebuilt. It makes vcpkg not very usable. Can you please atleast store prebuilt packages for common platforms e.g. x64-windows and for common packaged

**Environment**
-  x64-windows
- Compiler: VS 2022

**To Reproduce**
See description above

**Expected behavior**
vcpkg to be fast, like all other package managers like nuget, npm

**Failure logs** 
See description above

**Additional context**
See description above

**Comment by autoantwort** [comment]
According to https://github.com/microsoft/vcpkg/discussions/18332#discussioncomment-838933 they have not done this because of licensing issues

**Comment by JackBoosY** [comment]
1. vcpkg is a package manager that builds binary through port source code, so it will take more time than a package manager that directly provides binary.
2. You don't need to install the dependent port repeatedly, because the binary-caching can automatically restore the previously installed binary when the port is not changed.
3. > Can you please atleast store prebuilt packages for common platforms e.g. x64-windows and for common packaged

Maybe, vcpkg still doesn't provide precompiled binaries directly.

**Reply by mark-whitelaw** [reply]
2. is problematic when using github actions

**Comment by Darkster1** [comment]
+1.
Noticed in ProcessExplorer that vcpkg creates sub-processes with the lowest priority, changing the priority to "high" makes it fly.
Disadvantage of that is, when installing a lot of packages it still takes a few clicks to switch that priority, so maybe it's configurable somewhere?
Currently I'm running a lot of other processes though but noticed this issue at another time when I didn't have other processes running.

[
![vcpkg](https://user-images.githubusercontent.com/6094693/170141261-3d04613f-fcdb-4924-9cfa-85af10d93dcc.png)
](url)

**Comment by JackBoosY** [comment]
> Noticed in ProcessExplorer that vcpkg creates sub-processes with the lowest priority, changing the priority to "high" makes it fly.
Disadvantage of that is, when installing a lot of packages it still takes a few clicks to switch that priority, so maybe it's configurable somewhere?

I doubt how much higher process priority will speed up the whole process.

> Currently I'm running a lot of other processes though but noticed this issue at another time when I didn't have other processes running.

For 7zip, according to the [official document](http://7zip.bugaco.com/7zip/MANUAL/cmdline/switches/method.htm#ZipMultiThread), maybe we can add extra option `--mt=${VCPKG_CONCURRENCY}`
